runtime/compiler/control/J9CompilationStrategy.cpp

```diff
@@ -529,11 +646,13 @@ J9::CompilationStrategy::ProcessJittedSample::initializeRecompRelatedFields()
 
    if (_logSampling)
       {
-      _curMsg += sprintf(_curMsg, " cnt=%d ncl=%d glblSmplCnt=%d startCnt=%d[-%u,+%u] samples=[%d %d] windows=[%d %u] crtSmplIntrvlCnt=%u",
+      size_t maxMsgLen = sizeof(_msg) - (_curMsg - _msg);
+      size_t msgLen = snprintf(_curMsg, maxMsgLen, " cnt=%d ncl=%d glblSmplCnt=%d startCnt=%d[-%u,+%u] samples=[%d %d] windows=[%d %u] crtSmplIntrvlCnt=%u",
          _count, _methodInfo->getNextCompileLevel(), _totalSampleCount, _startSampleCount,
          _bodyInfo->getOldStartCountDelta(), _bodyInfo->getHotStartCountDelta(),
          _globalSamples, _globalSamplesInHotWindow,
          _scorchingSampleInterval, _hotSampleInterval, _crtSampleIntervalCount);
+      _curMsg += (msgLen <= maxMsgLen) ? msgLen : maxMsgLen;
```

Comment:
Similar adjustments are needed for this pattern (after declaring `msgLen` with type `int`):
```c
      if (msgLen > 0)
         _curMsg += ((size_t)msgLen <= maxMsgLen) ? msgLen : maxMsgLen;
```